python/ray/train/batch_predictor.py

```diff
@@ -260,11 +265,17 @@ def _keep_columns_from_input_batch(
                     return prediction_output_batch
 
             def __call__(self, input_batch: DataBatchType) -> DataBatchType:
+                # TODO: Delegate separate_gpu_stage flag to Datasets.
+                if override_prep:
```

Comment:
iiuc with this change is moving the separated stage preprocessor from dataset level to per scoring actor level ? do we have any data on the perf implications based on our existing benchmarks ?

Reply:
Not sure I understand.

It's the other way around-- we are moving the stage fusing decisions to datasets instead of manually implementing it in the scoring actor.

Reply:
in this PR, previously in case of `separate_gpu_stage and num_gpus_per_worker > 0` we have `preprocessor = self.get_preprocessor()` then it's `preprocessor._transform_batch` is executed by `data.map_batches` as UDF, in this PR the `transform_batch` is executed in the call of `ScoringWrapper`. 

In heterogenous cluster setup with n CPU workers and m GPU workers, assuming total parallelism is n + m at dataset, would this imply running transform batch in ScoringWrapper will have parallelism of m instead ? 

I might have missed some context here yet but trying to understand better how the stage fusion will happen here as it appears to be invoked further downstream (or maybe with lazy execution by default it won't matter where we call it anymore, thus every transform can be called locally ?)

Reply:
Ah this PR changes the condition from `if separate_gpu_stage` to `if **not** separate_gpu_stage`. The steps you described are happening in the else clause now.

Reply:
ah i see, this makes sense then :)